src/NServiceBus.Core/Unicast/Transport/TransportConnectionString.cs

```diff
@@ -56,7 +63,37 @@ static string ReadConnectionString(string connectionStringName)
 
   busConfig.UseTransport<{0}>().ConnectionString(""{1}"");
 ";
+    }
+#endif
 
-        const string DefaultConnectionStringName = "NServiceBus/Transport";
+#if NETSTANDARD2_0
+    sealed class TransportConnectionString
+    {
+        TransportConnectionString()
+        {
+            GetValue = () => null;
+        }
+
+        public TransportConnectionString(Func<string> func)
+        {
+            GetValue = func;
+        }
+
+        public static TransportConnectionString Default => new TransportConnectionString();
+
+        public string GetConnectionStringOrRaiseError(TransportDefinition transportDefinition)
+        {
+            var connectionString = GetValue();
+            if (connectionString == null && transportDefinition.RequiresConnectionString)
```

Comment:
I think technically we could provide different exception in case the user didn't configure a connection string function at all or whether the function returns null, but that's not something we need to address in this PR I think.

Reply:
Agreed